conda/base/context.py

```diff
@@ -187,6 +187,28 @@ class Context(Configuration):
     _croot = PrimitiveParameter('', aliases=('croot',))
     conda_build = MapParameter(string_types, aliases=('conda-build',))
 
+    def __init__(self, search_path=None, argparse_args=None):
+        if search_path is None:
+            search_path = SEARCH_PATH
+
+        if argparse_args:
+            # This block of code sets CONDA_PREFIX based on '-n' and '-p' flags, so that
+            # configuration can be properly loaded from those locations
+            func_name = ('func' in argparse_args and argparse_args.func or '').rsplit('.', 1)[-1]
+            if func_name in ('create', 'install', 'update', 'remove', 'uninstall', 'upgrade'):
+                if 'prefix' in argparse_args and argparse_args.prefix:
+                    os.environ['CONDA_PREFIX'] = argparse_args.prefix
+                elif 'name' in argparse_args and argparse_args.name:
```

Comment:
I see repeated use of:
```
if 'x' in argparse_args and argparse_args.x
```

There is absolutely nothing wrong with it. One could argue that it is more articulate. 

However, I would like your opinion on doing:
```
argparse_args_d = vars(argparse_args)
```
... and then using it as:
```
if argparse_args_d.get('x', None):
```

Reply:
Would be fine too.  I handled both cases in https://github.com/conda/conda/blob/d432501ea747202c8b1f6b6d32d131df59fdb4c6/conda/common/configuration.py#L767-L782

It would be creating an extra dict instance that's very soon thrown away though inside the `super` call I think.

Reply:
That `Namespace` thing that's part of argparse is just so dang weird.